src/main/java/org/terasology/changingBlocks/ChangingBlocksSystem.java

```diff
@@ -108,12 +101,10 @@ public void update(float delta) {
                                 changingBlocks.send(new OnBlockSequenceComplete());
                             }
                         }
-                        String newBlockUri = keyList.get(currentstageIndex);
+                        BlockUri newBlockUri = keyList.get(currentstageIndex);
                         Block newBlock = blockManager.getBlock(newBlockUri);
-                        if (newBlockUri.equals(newBlock.getURI().toString())) {
```

Comment:
`BlockManager#getBlock` cannot return `null`. If, before returning, the block that would be returned equals `null` the air block is returned instead.